src/services/users.js

```diff
@@ -0,0 +1,191 @@
+const { User } = require('../models');
+const { UserRoles } = require('../models');
+const { UserStatuses } = require('../models');
+const bcrypt = require('bcryptjs');
+
+class UsersService {
+  async createUser (req, res) {
+    try {
+      const userExist = await this.checkUserExists(req);
+      if (!userExist) {
+        const roleUser = await this.getRole('User');
+        const newUser = await this.createBasicUser(req);
+        newUser.role = roleUser._id;
+        await newUser.save();
+        return newUser ? res.json(true) : res.boom.notFound();
+      } else {
+        res.boom.badRequest('user already exists');
+      }
+    } catch (err) {
+      res.boom.badRequest();
+    }
+  }
+
+  async createAdmin (req, res) {
```

Comment:
createUser and createAdmin almost the same methods. There is only difference in role. So move this all logic into mutual for them createUser method, which accepts body and role

Reply:
Fixed